### PR TITLE
Pf/end to end

### PR DESCRIPTION
**Pull request by pfinashx**
Added convertation NNCF models to IR and test via Accuracy Checker

### CODE REVIEW

**Comment by vshampor** [comment]
Jenkins please retry a build

File: tests/test_sota_openvino.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+@pytest.mark.parametrize("model", model_list, ids=model_list)
+def test_convert_to_ir(model):
+    onnx_model = onnx_model_name(model)
+    subprocess.run("{} {} mo.py --input_model {} --framework=onnx --data_type=FP16 --reverse_input_channels"
+                   " --mean_values=[123.675,116.28,103.53] --scale_values=[58.4795,57.1429,57.4713]"
+                   " --output_dir {}".format(OPENVINO_ACTIVATE, sys.executable, onnx_model,
+                                             os.path.join(PROJECT_ROOT, "ir_models")),
+                   cwd=(os.path.join(OPENVINO_DIR, "deployment_tools", "model_optimizer")), check=True, shell=True)
```
Comment (vshampor) [inline tests/test_sota_openvino.py:134]: You separated different stages of one and the same test case into separate test cases, which is not necessarily wrong, but your IR conversion "test case" in fact should not be even run on its own without ONNX conversion, because it will definitely fail, and your `accuracy_checker` "test case" also cannot be run separately from the ONNX conversion and IR conversion. Either move all the stages into a single test case, or set up proper dependencies between the stages:
https://pytest-dependency.readthedocs.io/en/stable/usage.html

File: tests/test_sota_openvino.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+
+@pytest.mark.parametrize("model", model_list, ids=model_list)
+def test_accuracy_check(sota_data_dir, model):
+    subprocess.run("{} accuracy_check -c {}/{config}.yml -s {} -d dataset_definitions.yml --csv_result {config}.csv".
+                   format(OPENVINO_ACTIVATE, os.path.join(PROJECT_ROOT, "tests", "data", "ac_configs"), sota_data_dir,
+                          config=model), cwd=ACC_CHECK_DIR, check=True, shell=True)
```
Comment (vshampor) [inline tests/test_sota_openvino.py:141]: See above

**Review by vshampor — CHANGES_REQUESTED** [review]
General note - you are extending the current eval pipeline, not adding an entire new set of tests. 

If you stick with a "one stage - one test function approach", then group all the stages in a single test class with an appropriate name and set up dependencies between the stages.

Alternatively, you can bring the code from all the stages into a single test function and use flags and/or forwarded pytest arguments to determine which stages to run.

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Review by vshampor — CHANGES_REQUESTED** [review]

**Comment by vshampor** [comment]
Please also remove the "Merge commits" from this PR - rebase your fork's branch properly onto the https://github.com/openvinotoolkit/nncf_pytorch develop branch.

File: tests/conftest.py [hunk 3]
```diff
@@ -61,6 +64,9 @@ def pytest_addoption(parser):
     parser.addoption(
         "--third-party-sanity", action="store_true", default=False, help="To run third party sanity test cases"
     )
+    parser.addoption(
+        "--openvino", action="store_true", default=False, help="To run eval models via OpenVino"
```
Comment (vshampor) [inline tests/conftest.py:68]: Rename to `run_openvino_eval`

File: tests/test_sota_checkpoints.py [hunk 4]
```diff
@@ -23,6 +24,9 @@
 DIFF_FP32_MAX_GLOBAL = 0.1
 
 METRICS_DUMP_PATH = PROJECT_ROOT / 'metrics_dump'
+OPENVINO_DIR = os.path.join(os.path.dirname(PROJECT_ROOT), 'intel', 'openvino')
+ACC_CHECK_DIR = os.path.join(OPENVINO_DIR, 'deployment_tools', 'open_model_zoo', 'tools', 'accuracy_checker')
```
Comment (vshampor) [inline tests/test_sota_checkpoints.py:28]: Here, and from now on everywhere, use pathlib instead of os.path. The `METRICS_DUMP_PATH` is an example of such usage.

File: tests/test_sota_checkpoints.py [hunk 5]
```diff
@@ -108,6 +112,11 @@ def process_line(decoded_line: str, error_lines: List):
         err_string = "\n".join(error_lines) if error_lines else None
         return exit_code, err_string
 
+    @staticmethod
+    def onnx_model_name(name):
```
Comment (vshampor) [inline tests/test_sota_checkpoints.py:116]: Wrong function naming again. `get_onnx_model_file_path`

File: tests/test_sota_checkpoints.py [hunk 6]
```diff
@@ -327,6 +340,64 @@ def test_eval(self, sota_checkpoints_dir, sota_data_dir, eval_test_struct: EvalR
...
+            onnx_cmd += " --pretrained"
+        exit_code, err_str = self.run_cmd(onnx_cmd, cwd=PROJECT_ROOT)
+        if exit_code != 0:
+            self.row_dict[eval_test_struct.model_name_].append(err_str)
+            self.color_dict[eval_test_struct.model_name_] = BG_COLOR_RED_HEX
+            assert False
+
+        onnx_model = self.onnx_model_name(eval_test_struct.model_name_)
+        mo_cmd = "{} mo.py --input_model {} --framework=onnx --data_type=FP16 --reverse_input_channels" \
+                 " --mean_values=[123.675,116.28,103.53] --scale_values=[58.4795,57.1429,57.4713] --output_dir {}"\
```
Comment (vshampor) [inline tests/test_sota_checkpoints.py:375]: `--mean_values` and `--scale_values` is something that is set up per-model, not globally. For some models, it may even be unnecessary.

File: tests/test_sota_checkpoints.py [hunk 7]
```diff
@@ -359,12 +430,25 @@ def skip_params(sota_data_dir):
 
 
 @pytest.fixture(autouse=True, scope="class")
-def results():
+def openvino_preinstall(openvino):
+    if openvino is not None:
+        subprocess.run("pip uninstall setuptools -y && pip install setuptools", check=True, shell=True)
```
Comment (vshampor) [inline tests/test_sota_checkpoints.py:435]: Why is this required?

**Review by vshampor — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 3 above)
Comment (pfinashx) [reply tests/conftest.py:68]: Done

File: tests/test_sota_checkpoints.py (hunk 4 above)
Comment (pfinashx) [reply tests/test_sota_checkpoints.py]: Done

File: tests/test_sota_checkpoints.py (hunk 5 above)
Comment (pfinashx) [reply tests/test_sota_checkpoints.py]: Done

File: tests/test_sota_checkpoints.py (hunk 6 above)
Comment (pfinashx) [reply tests/test_sota_checkpoints.py:375]: Are we going to take these values ​​from the sota_checkpoints_eval.json file? Or from another?

File: tests/test_sota_checkpoints.py (hunk 7 above)
Comment (pfinashx) [reply tests/test_sota_checkpoints.py:434]: Without this, an AC installation error occurs. Not sure it's CI docker problem or setuptools

Comment (vshampor) [reply tests/test_sota_checkpoints.py:434]: Add a comment in code explaining this.

File: tests/test_sota_checkpoints.py (hunk 6 above)
Comment (vshampor) [reply tests/test_sota_checkpoints.py:375]: Yes, let's use the `sota_checkpoints_eval.json` to supply this information for now.

**Comment by pfinashx** [comment]
Latest results - https://adas-ci-icv.inn.intel.com/job/NNCF/job/manual/job/ubuntu16/87/

**Comment by pfinashx** [comment]
Rebased on last develop, started Build #89

**Review by vshampor — APPROVED** [review]

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by pfinashx** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build
